### PR TITLE
Fix query, tests and ci checks

### PR DESCRIPTION
**Pull request by SD-13**
In `oppia/oppia`, this action failed for wrong graphql query and this PR fix that query, relevant tests, and the ci checks of this repo.

For context: [This job ](https://github.com/oppia/oppia/actions/runs/5823413288/job/15790246086) failed because of using wrong query. Previously, the query was fetching all the discussions instead of discussions in a particular category. This PR breaks that step into two. Now, first, it is fetching the existing categories, then after filtering the category id of a particular category name, it is using another query to get the discussions in that particular category.

### CODE REVIEW

**Comment by SD-13** [comment]
@seanlip @U8NWXD Sorry for the inconvenience, please take a look at the changes. Thanks!

**Review by seanlip — APPROVED** [review]
@SD-13 Thanks -- no particular concerns from my end. I'll defer to you and @U8NWXD about whether he needs to look at these changes before it gets merged.

**Pull request merged**
